filesystem/fat32/fat32_test.go

```diff
@@ -1058,3 +1070,133 @@ func TestOpenFileCaseInsensitive(t *testing.T) {
 		}
 	}
 }
+
+// small helper function to make a large fat32 filesystem
+func mkfs(name string) (filesystem.FileSystem, error) {
+	size := int64(6 * 1024 * 1024 * 1024)
+	d, err := diskfs.Create(name, size, diskfs.Raw, diskfs.SectorSizeDefault)
+	if err != nil {
+		fmt.Printf("error creating disk: %v", err)
+		os.Exit(1)
+	}
+
+	spec := disk.FilesystemSpec{
+		Partition: 0,
+		FSType:    filesystem.TypeFat32,
+	}
+
+	fs, err := d.CreateFilesystem(spec)
+	return fs, err
+}
+
+func mkdir(fs filesystem.FileSystem, name string) error {
+	err := fs.Mkdir(name)
+	return err
+}
+
+func mkMicroFile(fs filesystem.FileSystem, name string) error {
+	rw, err := fs.OpenFile(name, os.O_CREATE|os.O_RDWR)
+	if err != nil {
+		return err
+	}
+
+	_, err = rw.Write([]byte("hello World"))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func mkSmallFile(fs filesystem.FileSystem, name string) error {
+	rw, err := fs.OpenFile(name, os.O_CREATE|os.O_RDWR)
+	if err != nil {
+		return err
+	}
+
+	size := 5 * 1024 * 1024
+	smallFile := make([]byte, size)
+	_, err = rw.Write(smallFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func mkRandFile(fs filesystem.FileSystem, name string, rSize int32) error {
+	rw, err := fs.OpenFile(name, os.O_CREATE|os.O_RDWR)
+	if err != nil {
+		return err
+	}
+
+	size := rSize * 1024 * 1024
+	randFile := make([]byte, size)
+	_, err = rw.Write(randFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func mkGigFile(fs filesystem.FileSystem, name string) error {
+	rw, err := fs.OpenFile(name, os.O_CREATE|os.O_RDWR)
+	if err != nil {
+		return err
+	}
+
+	size := 1024 * 1024 * 1024
+	smallFile := make([]byte, size)
+	_, err = rw.Write(smallFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func TestCreateFileTree(t *testing.T) {
+	fileName := "file-tree.img"
```

Comment:
Can you move this to a tempfile? Something like:

```go
	tmpDir := t.TempDir()
	tmpImgPath := filepath.Join(tmpDir, "file-tree.img")
```

That way it does not pollute the code tree, gets cleaned up automatically at the end of the test, and even if that gets missed, eventually the OS cleans up those tmp dirs.